### ISSUE TITLE
[BUG] redis WARN: CROSSSLOT 

### ISSUE THREAD

**Issue by ChnMig** [hibiken/asynq#480]
**Describe the bug**
When I set redis to cluster mode and connect using `asynq.RedisClusterClientOpt`, `asynq` reports frequent errors:`WARN: recoverer: could not move task to archive: INTERNAL_ERROR: redis eval error: CROSSSLOT Keys in request don't hash to the same slot`

**To Reproduce**
When he starts, it will definitely reappear

**Environment (please complete the following information):**
 - OS: [e.g. MacOS, Linux]
 - Version of `asynq` package 0.23.0

**Comment by hibiken** [comment]
@ChnMig Thank you for reporting an issue.
Based on the error message you provided, it seems that error is returned from this call to `Archive` (https://github.com/hibiken/asynq/blob/6c954c87bf8c1aaca55ab824b47125d2359eea91/recoverer.go#L123-L125).

If you look at the implementation of the `Archive`, it should all have the same hash tag of `{qname}` https://github.com/hibiken/asynq/blob/6c954c87bf8c1aaca55ab824b47125d2359eea91/internal/rdb/rdb.go#L824-L867)

Do you have queue name or TaskID  with `{}` in the string?

**Comment by ChnMig** [comment]
@hibiken Ok, I have changed to standalone redis for now, but I will try it in my free time, I will close this issues first, when I have conclusion, I will let you know here, thanks for your help

**Comment by misterGF** [comment]
I experienced a similar issue. My error was  **WARN: recoverer: could not move task to archive: INTERNAL_ERROR: redis eval error: NOT FOUND**

This happened after I mucked up a record in redis. I was trying to edit it manually and something went awry.

I fixed it by looking at my keys for something that seemed different from my staging environment. 

I spotted a key that existed in the staging environment

**asynq:{default}:active** | ["0c8ff109-6d22-4dcf-aedb-749e90c96e94"]

I ended up renaming the key name to "a" to test if it it would fix it and it did!

Probably could have fixed it by removing that id in the value as well. Hope this helps someone.
